testutil/integration/evmos/factory/build.go

```diff
@@ -76,3 +77,81 @@ func (tf *IntegrationTxFactory) GenerateSignedEthTx(privKey cryptotypes.PrivKey,
 
 	return tf.buildSignedTx(signedMsg)
 }
+
+// GenerateMsgEthereumTx creates a new MsgEthereumTx with the provided arguments.
+// If any of the arguments are not provided, they will be populated with default values.
+func (tf *IntegrationTxFactory) GenerateMsgEthereumTx(
+	privKey cryptotypes.PrivKey,
+	txArgs evmtypes.EvmTxArgs,
+) (evmtypes.MsgEthereumTx, error) {
+	fromAddr := common.BytesToAddress(privKey.PubKey().Address().Bytes())
+	// Fill TxArgs with default values
+	txArgs, err := tf.populateEvmTxArgs(fromAddr, txArgs)
+	if err != nil {
+		return evmtypes.MsgEthereumTx{}, errorsmod.Wrap(err, "failed to populate tx args")
+	}
+	msg := buildMsgEthereumTx(txArgs, fromAddr)
+
+	return msg, nil
+}
+
+// GenerateGethCoreMsg creates a new GethCoreMsg with the provided arguments.
+func (tf *IntegrationTxFactory) GenerateGethCoreMsg(
+	privKey cryptotypes.PrivKey,
+	txArgs evmtypes.EvmTxArgs,
+) (core.Message, error) {
+	msg, err := tf.GenerateMsgEthereumTx(privKey, txArgs)
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "failed to generate ethereum tx")
+	}
+
+	signedMsg, err := tf.SignMsgEthereumTx(privKey, msg)
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "failed to sign ethereum tx")
+	}
+
+	baseFeeResp, err := tf.grpcHandler.GetBaseFee()
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "failed to get base fee")
+	}
+	signer := gethtypes.LatestSignerForChainID(
+		tf.network.GetEIP155ChainID(),
+	)
+	return signedMsg.AsMessage(signer, baseFeeResp.BaseFee.BigInt())
+}
```

Comment:
what is the reason for adding this method? It doesn't seem to be used anywhere, is this for the SDK v0.50 logic?

Reply:
There are certain areas in the code where the Core Msg is needed. I took it from the sdk 50 PR, where unit tests have been refractored with the newer integration setup to avoid conflicts in the future. Those refractos have nothing to do with the sdk50 but just legacy tests that had not been updated yet.